src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { useTodos } from '../../hooks/useTodos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+};
+
+const TodoItem: React.FC<Props> = ({ todo }) => {
+  const [selectedTodoId, setSelectedTodoId] = React.useState<number>(0);
+
+  const { todos, setTodos, isLoading, isAllDeleted, onDeleteTodo } = useTodos();
+
+  const handleCheckbox = () => {
+    setTodos(
+      todos.map(prevTodo => {
+        return prevTodo.id === todo.id
+          ? { ...prevTodo, completed: !prevTodo.completed }
+          : prevTodo;
+      }),
+    );
+  };
+
+  const handleDeleteTodo = () => {
+    setSelectedTodoId(todo.id);
+    onDeleteTodo(todo.id);
+  };
+
+  // eslint-disable-next-line prettier/prettier
+  const isLoaderActive = (isLoading && todo.id === selectedTodoId)
+  || (todo.completed && isAllDeleted);
```

Comment:
That is strange and too complicated logic and these conditions will be more. You can create a state where you can store ids of todos that currently loading and depending on it set a loader.